Assets/Scripts/Obstacles/ExplodingBarrel.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class ExplodingBarrel : MonoBehaviour
+{
+
+    [SerializeField]
+    private MeshRenderer barrelMesh;
+
+    void Start()
+    {
+        GetComponent<HealthController>().onDeath += Explode;
+    }
+
+    private void Explode(HealthController controller, float damage, DamageInfo info)
+    {
+        barrelMesh.enabled = false;
+        GetComponentInChildren<CapsuleCollider>().enabled = false;
+        GetComponent<ExplosionController>().Explode(info.sourcePlayer);
+        Destroy(gameObject, 4);
+    }
+}
+
+//Husk hvem som gjorde skade og send det videre
```

Comment:
Should maybe rewrite to english or delete

Reply:
whoops, there was something more I'd missed here :sweat_smile: